docker/maistra-builder_2.0.Dockerfile

```diff
@@ -1,3 +1,12 @@
+# Hack: Remove this once the base image has go 1.16
+FROM golang:1.16.4 AS go116
+
+ENV K8S_TEST_INFRA_VERSION=03cf33ddeb
+RUN git clone https://github.com/kubernetes/test-infra.git /root/test-infra && \
+    cd /root/test-infra && git checkout ${K8S_TEST_INFRA_VERSION} && \
+    go build -o /usr/local/bin/checkconfig prow/cmd/checkconfig/main.go && \
+    go build -o /usr/local/bin/pr-creator robots/pr-creator/main.go
+
 FROM fedora:32
```

Comment:
@jwendell Now that we are bumping the go version. Are there any additional steps required to update this to `fedora:34`?

Reply:
We are not bumping go version. I'm using go 1.16 to build *only* these two tools, that are related to prow/automation, not with istio itself.